test/models/button_hold_recall_test.rb

```diff
@@ -0,0 +1,164 @@
+  setup do
+    VCR.use_cassette('realtime aleph volume') do
+      @item = AlephItem.new.xml_status('MIT01001019412').xpath('//items').children.first
+    end
+    @oclc = '123456789'
+    @scan = 'true'
+    @button = ButtonHoldRecall.new(@item, @oclc, @scan)
+  end
+
+  test 'html_button_recall' do
+    @button.stub :eligible_recall?, true do
+      assert(@button.html_button.include?('Recall (7+ days)'))
+    end
+  end
+
+  test 'html_button_hold' do
+    @button.stub :eligible_hold?, true do
+      assert(@button.html_button.include?('Place hold (1-2 days)'))
+    end
+  end
+
+  test 'html_button_ineligible' do
+    @button.stub :eligible_hold?, false do
+      assert_nil(@button.html_button)
+    end
+  end
+
+  test 'eligible for hold' do
+    maker = ButtonHoldRecall.new(@item, @oclc, @scan)
+    maker.instance_variable_set(:@status, 'In Library')
+    # todo: hold_recallable is no longer an instance variable. refactor tests.
+    maker.instance_variable_set(:@hold_recallable, true)
+    assert maker.eligible_hold?
+
+    maker.instance_variable_set(:@status, 'MIT Reads')
+    assert maker.eligible_hold?
+
+    # todo: hold_recallable is no longer an instance variable. refactor tests.
+    # maker.instance_variable_set(:@hold_recallable, false)
+    # refute maker.eligible?
+
+    maker.instance_variable_set(:@on_reserve, true)
+    refute maker.eligible_hold?
+
+    maker.instance_variable_set(:@on_reserve, false)
+    maker.instance_variable_set(:@library, 'Physics Dept. Reading Room')
+    refute maker.eligible_hold?
+  end
+
+  test 'eligible for recall' do
+    maker = ButtonHoldRecall.new(@item, @oclc, @scan)
+    maker.stub :hold_recallable?, true do
+      maker.instance_variable_set(:@status, 'Due sometime')
+      assert maker.eligible_recall?
+
+      maker.instance_variable_set(:@on_reserve, true)
+      refute maker.eligible_recall?
+
+      maker.instance_variable_set(:@on_reserve, false)
+      maker.instance_variable_set(:@library, 'Physics Dept. Reading Room')
+      refute maker.eligible_recall?
+    end
+  end
+
+  test 'eligible for recall edge case' do
+    maker = ButtonHoldRecall.new(@item, @oclc, @scan)
+    # todo: hold_recallable is no longer an instance variable. refactor tests.
+    maker.instance_variable_set(:@hold_recallable, true)
+    maker.instance_variable_set(:@status, 'Missing')
+    refute maker.eligible_recall?
+  end
+
+  test 'hold ineligibility by reason of z30 status code' do
+    maker = ButtonHoldRecall.new(@item, @oclc, @scan)
+
+    # Check our assumption that the default ButtonMaker can be held/recalled.
+    assert maker.eligible_hold?
+
+    # Now test z30 status codes that should yield ineligibility.
+    maker.instance_variable_set(:@z30status_code, '04')
+    # We need to reset @hold_recallable as it was set on object initialization, but
+    # the change to the z30 status will alter its expected value.
+    # todo: hold_recallable is no longer an instance variable. refactor tests.
```

Comment:
@thatandromeda this is one example of the tests that really should probably fail. I haven't thought much about it as I wanted to get feedback on the general approach of this refactor before coming back and cleaning up the few bits that are just weird.

Reply:
In other words, I am fairly confident they are no longer actually testing anything useful. I plan to dig in on what they were testing and write new tests to cover those situations.

Reply:
👍

Reply:
This looks right? If the code is `04`, `hold_recallable?` should be `false`, so  `eligible_hold?` should be false.

But I feel fine about rethinking what they're actually testing and updating accordingly. At any rate, I don't see anything obviously weird going on here.